### PR TITLE
Swift object module

### PR DESCRIPTION
**Pull request by Polina-Gubina**
Swift object module

### CODE REVIEW

**Comment by Polina-Gubina** [comment]
recheck

File: tests/integration/targets/object/tasks/main.yaml [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+    - name:
+      set_fact:
+        container_name: "test-container"
+        first_object_name: "first-object"
+        second_object_name: "second-object"
+        path_file: ./
+        content_file: testfile.txt
+        dest: "./dest.txt"
+
+    - name: Create file with content
```
Comment (anton-sidelnikov) [inline tests/integration/targets/object/tasks/main.yaml:15]: create several different files

File: tests/integration/targets/object/tasks/main.yaml [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+      assert:
+        that:
+          - container is success
+          - container is changed
+
+    - name: Create second object
+      opentelekomcloud.cloud.object:
+        container: "{{ container_name }}"
+        object: "{{ second_object_name }}"
+        content: "{{ content_file }}"
```
Comment (anton-sidelnikov) [inline tests/integration/targets/object/tasks/main.yaml:44]: it works? or it should be "{{ path_file }}/{{ content_file }}"? please check

Comment (anton-sidelnikov) [reply tests/integration/targets/object/tasks/main.yaml:44]: should be abs path

**Review by anton-sidelnikov — CHANGES_REQUESTED** [review]

**Comment by Polina-Gubina** [comment]
recheck

**Review by anton-sidelnikov — DISMISSED** [review]

**Review by lego963 — DISMISSED** [review]

File: plugins/modules/object.py [hunk 3]
```diff
@@ -0,0 +1,403 @@
...
+                self.exit_json(changed=True, **data)
+            self.exit_json(changed=False)
+
+        if not self._container_exist(container):
+            if self.params['ignore_nonexistent_container']:
+                container_data = self.conn.object_store.create_container(name=container).to_dict()
+                container_data.pop('location')
+                data['container'] = container_data
+            else:
+                self.fail_json("Container doesn't exist")
```
Comment (outcatcher) [inline plugins/modules/object.py:257]: Usages of `self._container_exist(container)` and `self.conn.object_store.create_container` are duplicating. Can this be simplified?

**Review by outcatcher — CHANGES_REQUESTED** [review]

**Comment by Polina-Gubina** [comment]
recheck

**Comment by Polina-Gubina** [comment]
recheck

**Review by anton-sidelnikov — APPROVED** [review]

**Review by lego963 — APPROVED** [review]

**Review by outcatcher — APPROVED** [review]

**Review by gtema — CHANGES_REQUESTED** [review]
1) Please separate container management into a separate module
2) Please change object upload from file to be using "source" or "src" attribute and not the "content". This can avoid lots of potential corner cases and is in sync with other ansible modules
3) If we develop it in openstack directly we can get this merged pretty quickly (but I will not insist on that, it is just effort saving).

**Comment by gtema** [comment]
and then also, why do you try to compete with openstack implementation?

**Pull request closed without merging**